src/Novell.Directory.Ldap.NETStandard/Rfc2251/RfcFilter.cs

```diff
@@ -1052,29 +1052,23 @@ private static void StringFilter(IEnumerable<object> filterEnumerable, StringBui
         ///     Convert a UTF8 encoded string, or binary data, into a String encoded for
         ///     a string filter.
         /// </summary>
-        private static string ByteString(byte[] valueRenamed)
+        private static string ByteString(byte[] value)
         {
-            if (Base64.IsValidUtf8(valueRenamed, true))
+            if (Base64.IsValidUtf8(value, true))
             {
-                return valueRenamed.ToUtf8String();
+                return value.ToUtf8String();
             }
 
-            var binary = new StringBuilder();
-            for (var i = 0; i < valueRenamed.Length; i++)
+            var binary = new StringBuilder(value.Length * 3);
+            for (var i = 0; i < value.Length; i++)
             {
-                // TODO repair binary output
                 // Every octet needs to be escaped
-                if (valueRenamed[i] >= 0)
-                {
-                    // one character hex string
-                    binary.Append("\\0");
-                    binary.Append(Convert.ToString(valueRenamed[i], 16));
-                }
-                else
-                {
-                    // negative (eight character) hex string
-                    binary.Append("\\" + Convert.ToString(valueRenamed[i], 16).Substring(6));
-                }
+                binary.Append('\\');
+#if NET6_0_OR_GREATER

```

Comment:
do we want to have an if just for a bit nicer syntax?

Reply:
It also means, that `value[i]` is not boxed as `object`, but otherwise it should behave the same and both should use the fact, that `byte` implements `ISpanFormattable` in runtime-versions, that support it. Using always the `$""` would sadly mean, that before .NET6 we allocate a lot of short-lived strings.